src/ruby/yast/debugger.rb

```diff
@@ -0,0 +1,157 @@
+          if UI.TextMode || remote || !start_client
+            # in textmode or in remote mode ask the user to start
+            # the debugger client manually
+            UI.OpenDialog(Label(debugger_message(remote, port)))
+            popup = true
+          else
+            # in GUI open an xterm session with the debugger
+            start_gui_session(port)
+            # wait a bit to get the server ready (to avoid "Broken pipe" error)
+            # FIXME: looks like a race condition inside byebug itself...
+            wait = true
+          end
+
+          # start the server and wait for connection
+          start_server(remote, port, wait)
+
+          UI.CloseDialog if popup
+        end
+
+        # start the debugger session
+        byebug
+        # Now you can inspect the current state in the debugger,
+        # or use "next" to continue.
+        # Use "help" command to see the available commands, see more at
+        # https://github.com/deivid-rodriguez/byebug/blob/master/GUIDE.md
+      end
+
+      # is the Ruby debugger installed and can be loaded?
+      # @return [Boolean] true if the debugger is present
+      def installed?
+        require "byebug"
+        true
+      rescue LoadError
+        false
+      end
+
+    private
+
+      # starts the debugger server and waits for a cleint connection
+      # @param [Boolean] remote if set to true the server is accesible from network
+      # @param [Fixnum] port the port number used by the server
+      def start_server(remote, port, wait)
+        Byebug.wait_connection = true
+        host = remote ? "0.0.0.0" : "localhost"
+        log.info "Starting debugger server (#{host}:#{port}), waiting for connection..."
+        Byebug.start_server(host, port)
+        sleep(3) if wait
+      end
+
+      # starts a debugger session in xterm
+      # @param [Fixnum] port the port number to connect to
+      def start_gui_session(port)
+        job = fork do
+          # wait until the main thread starts the debugger and opens the port
+          # for listening
+          loop do
+            break if port_open?(port)
+            sleep(1)
+          end
+
+          # start the debugger client in an xterm session
+          exec "xterm", "-e", "byebug", "-R", "#{port}"
+        end
+
+        # detach the process, we do not wait for it so avoid zombies
+        Process.detach(job)
+      end
+
+      # compose the popup message describing how to manually connect to
+      # the running debugger
+      # @return [String] text
+      def debugger_message(remote, port)
+        waiting = "Waiting for the connection..."
+        if remote
+          # get the local IP addresses
+          require "socket"
+          remote_ips = Socket.ip_address_list.select { |a| a.ipv4? && !a.ipv4_loopback? }
+          cmds = remote_ips.map { |a| debugger_cmd(a.ip_address, port) }.join("\n")
+
+          "To connect to the debugger from a remote machine use this command:" \
```

Comment:
Exactly, I have omitted the `_()` translation marks as this is expert feature and will be used mainly by the YaST team members. I do not expect much usage by real customers, maybe remotely assisted debugging for L3s :wink: